rules-tests/DeadCode/Rector/FunctionLike/NarrowTooWideReturnTypeRector/Fixture/edge_cases.php.inc

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Rector\Tests\DeadCode\Rector\FunctionLike\NarrowTooWideReturnTypeRector\Fixture;
+
+final class EdgeCases
+{
+    public function implicitNull(): string|int|null
+    {
+        if (rand(0, 1)) {
+            return;
+        }
+
+        return 'something';
+    }
+
+    public function implicitReturn(): string|int|null
+    {
+        if (rand(0, 1)) {
+            return 'something';
+        }
+    }
+
+    public function noReturn(): string|int|null
+    {
+        if (rand(0, 1)) {
+            echo 'something';
+        }
+    }
+
+    public function noReturnWithImplicitMixed()
+    {
+        if (rand(0, 1)) {
+            echo 'something';
+        }
+    }
```

Comment:
please skip this, there is already rule for it

https://getrector.com/demo/e2bbdb3f-4cf9-4c93-b705-4ae993afb7c9

please only apply on:

- nullable
- union

Reply:
There is already rule for it, see https://getrector.com/demo/e2bbdb3f-4cf9-4c93-b705-4ae993afb7c9

Please only apply on these condition:

- only when functionlike has native return type
- only when native return type is nullable or union
- otherwise, just skip :)

Reply:
Also, if this new rule does the same thing as those other rules (just with the absence of returning early if there's no return type), then maybe those other rules should just be replaced by this?

Reply:
That already error on very first usage, see https://3v4l.org/2Bso3

Rector is not error handler :)

Reply:
Maybe not, but it's a free lunch and could help the user correct the return type if they made a mistake.

However, I've updated  :+1: